<?xml version="1.0"?>
<launch>
    <arg name="robot_ip" />
    <arg name="arm_id" default="panda" />
    <arg name="transmission" default="effort"
        doc="The type of position control to use (either 'position' or 'effort')" />
    <arg name="load_gripper" default="false"/> 

    <!-- activate franka control, enable error recovery action, load urdf, spawn FrankaStateController, robot_state_publisher, joint_state_publisher -->
    <include file="$(find panda_essentials)/launch/franka_control.launch" pass_all_args="true">
        <!-- <arg name="load_gripper" value="false" /> -->
    </include>

    <node name="controller_spawner"
        pkg="controller_manager"
        type="spawner"
        respawn="false"
        output="screen"
        args="$(arg transmission)_joint_trajectory_controller">
    </node>

    <node name="any_joint_pose"
        pkg="panda_essentials"
        type="any_joint_pose.py"
        output="screen"
        required="true">
        <!-- load initial pose to ~/joint_pose -->
        <rosparam file="$(find panda_essentials)/config/start_pose.yaml" subst_value="true" />
        <remap from="~follow_joint_trajectory"
            to="$(arg transmission)_joint_trajectory_controller/follow_joint_trajectory" />
        <remap from="~joint_states" to="franka_state_controller/joint_states" />
        <param name="max_dq" value="0.2" /> <!-- [rad/s] -->
    </node>

</launch>